debian/marquee/nginx/conf/http.conf

```diff
@@ -0,0 +1,6 @@
+server {
+    listen 80 default_server;
+    listen [::]:80 default_server;
+    server_name _;
```

Comment:
Is this needed?

Reply:
server_name _;

> Is this needed?

I’m actually not sure. The [nginx docs say](https://nginx.org/en/docs/http/server_names.html):

> If no server_name is defined in a server block then nginx uses the empty name as the server name.

So I’ve always the convention of setting the name to `_` and can say from experience I know that’s never caused any problems—but can’t say I know from experience I know that leaving it unset so the “empty name” gets used *doesn’t* cause any problems.

But anyway I’ll test it and see.

Reply:
server_name _;

>> Is this needed?
>
> I’ll test it and see.

I’ve now tested and can’t find that it causes any problems. From reading the docs and thinking about it, I guess the only thing it would ever affect is requests that don’t send the `Host` header. So that’s something we don’t need to care about anyway—since no browsers ever do that, and it’s non-conforming anyway (in HTTP 1.1 and 2 at least). And nginx itself rejects such requests with a 400 Bad Request before ever evaluating against the config.